pkg/api/login_oauth.go

```diff
@@ -80,6 +81,16 @@ func (hs *HTTPServer) OAuthLogin(ctx *models.ReqContext) {
 			return
 		}
 
+		stateConfig := setting.OAuthService.OAuthInfos[name].State
```

Comment:
Use non-global config:

```suggestion
		stateConfig := hs.Cfg.OAuthService.OAuthInfos[name].State
```

Reply:
done